src/System.Security.Cryptography.Xml/tests/Samples/EncryptingDecryptingAsymmetric.cs

```diff
@@ -0,0 +1,103 @@
+<test>some text node</test>
+</example>";
+
+        private static XmlDocument LoadXmlFromString(string xml)
+        {
+            var doc = new XmlDocument();
+            doc.PreserveWhitespace = true;
+            doc.LoadXml(xml);
+            return doc;
+        }
+
+        private static void Encrypt(XmlDocument doc, string elementName, string encryptionElementID, RSA rsaKey, string keyName)
+        {
+            var elementToEncrypt = (XmlElement)doc.GetElementsByTagName(elementName)[0];
+
+            using (var sessionKey = Aes.Create())
+            {
+                sessionKey.KeySize = 256;
+
+                // Encrypt the session key and add it to an EncryptedKey element.
+                var encryptedKey = new EncryptedKey()
+                {
+                    CipherData = new CipherData(EncryptedXml.EncryptKey(sessionKey.Key, rsaKey, false)),
+                    EncryptionMethod = new EncryptionMethod(EncryptedXml.XmlEncRSA15Url)
+                };
+
+                // Specify which EncryptedData
+                // uses this key. An XML document can have
+                // multiple EncryptedData elements that use
+                // different keys.
+                encryptedKey.AddReference(new DataReference()
+                {
+                    Uri = "#" + encryptionElementID
+                });
+
+                var encryptedData = new EncryptedData()
+                {
+                    Type = EncryptedXml.XmlEncElementUrl,
+                    Id = encryptionElementID,
+
+                    // Create an EncryptionMethod element so that the
+                    // receiver knows which algorithm to use for decryption.
+                    EncryptionMethod = new EncryptionMethod(EncryptedXml.XmlEncAES256Url)
+                };
+
+                encryptedData.KeyInfo.AddClause(new KeyInfoEncryptedKey(encryptedKey));
+                encryptedKey.KeyInfo.AddClause(new KeyInfoName()
+                {
+                    Value = keyName
+                });
+
+                var encryptedXml = new EncryptedXml();
+                encryptedData.CipherData.CipherValue = encryptedXml.EncryptData(elementToEncrypt, sessionKey, false);
+
+                EncryptedXml.ReplaceElement(elementToEncrypt, encryptedData, false);
+            }
+        }
+
+        public static void Decrypt(XmlDocument doc, RSA rsaKey, string keyName)
+        {
+            var encrypted = new EncryptedXml(doc);
+            encrypted.AddKeyNameMapping(keyName, rsaKey);
+            encrypted.DecryptDocument();
+        }
+
+        [Fact]
+        public void AsymmetricEncryptionRoundtrip()
+        {
+            using (RSA key = RSA.Create())
+            {
+                XmlDocument xmlDocToEncrypt = LoadXmlFromString(ExampleXml);
+                Encrypt(xmlDocToEncrypt, ExampleXmlRootElement, "EncryptedElement1", key, "rsaKey");
+
+                Assert.DoesNotContain("some text node", xmlDocToEncrypt.OuterXml);
```

Comment:
I'd put this as a const string up by the input text so that if one changes, so might the other.

Probably also good to assert that xmlDocToEncrypt.OuterXml DOES contain the value before encrypting it, so that if someone would notice that the post-encrypt assert has a chance of failing.